static/js/input_pill.js

```diff
@@ -96,23 +97,30 @@ export function create(opts) {
 
             store.pills.push(payload);
 
-            const has_image = item.img_src !== undefined;
+            const has_image = payload.item.img_src !== undefined;
 
             const opts = {
                 id: payload.id,
-                display_value: item.display_value,
+                display_value: payload.item.display_value,
+                deactivated_user_pill: false,
                 has_image,
             };
 
             if (has_image) {
-                opts.img_src = item.img_src;
+                opts.img_src = payload.item.img_src;
             }
 
             if (typeof store.onPillCreate === "function") {
                 store.onPillCreate();
             }
 
+            if (item.type === "user" && !people.is_person_active(item.user_id)) {
+                opts.deactivated_user_pill = true;
+                opts.display_value = payload.item.display_value + " (deactivated)";
```

Comment:
I think this belongs in `user_pill.js` somewhere, not in this file, right?

Reply:
I need to understand more of a bit why that PR added this check here. I just updated the check to be more better. I will see and let you know.

Reply:
Yes, so I researched more about this. I understood the whole code how `input_pills.js` is just for handling pills and we pass different functions in it. I debugged and found the right place to add it. Add this code in `user_pill.js`

Reply:
Now updating the test is left for commit 2.